service/src/main/java/bio/terra/workspace/service/workspace/model/AwsCloudContextFields.java

```diff
@@ -0,0 +1,166 @@
+    this.majorVersion = majorVersion;
+    this.organizationId = organizationId;
+    this.accountId = accountId;
+    this.tenantAlias = tenantAlias;
+    this.environmentAlias = environmentAlias;
+  }
+
+  public String getMajorVersion() {
+    return majorVersion;
+  }
+
+  public String getOrganizationId() {
+    return organizationId;
+  }
+
+  public String getAccountId() {
+    return accountId;
+  }
+
+  public String getTenantAlias() {
+    return tenantAlias;
+  }
+
+  public String getEnvironmentAlias() {
+    return environmentAlias;
+  }
+
+  public ApiAwsContext toApi() {
+    return new ApiAwsContext()
+        .majorVersion(majorVersion)
+        .organizationId(organizationId)
+        .accountId(accountId)
+        .tenantAlias(tenantAlias)
+        .environmentAlias(environmentAlias);
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (!(o instanceof AwsCloudContextFields that)) return false;
+    return Objects.equal(majorVersion, that.majorVersion)
+        && Objects.equal(organizationId, that.organizationId)
+        && Objects.equal(accountId, that.accountId)
+        && Objects.equal(tenantAlias, that.tenantAlias)
+        && Objects.equal(environmentAlias, that.environmentAlias);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(majorVersion, organizationId, accountId, tenantAlias, environmentAlias);
+  }
+
+  /**
+   * Verifies that current cloud context matches the current environment
+   *
+   * @param environment discovery environment
+   * @throws StaleConfigurationException StaleConfigurationException if they do not match
+   */
+  public void verifyCloudContext(Environment environment) {
+    Metadata metadata = environment.getMetadata();
+    if (!majorVersion.equals(metadata.getMajorVersion())
+        || !accountId.equals(metadata.getAccountId())) {
+      throw new StaleConfigurationException(
+          String.format(
+              "AWS cloud context mismatch. Metadata version is %s, expected %s; Account id is %s, expected %s",
+              majorVersion, metadata.getMajorVersion(), accountId, metadata.getAccountId()));
+    }
+  }
+
+  public String serialize() {
+    AwsCloudContextV1 dbContext = new AwsCloudContextV1(this);
+    return DbSerDes.toJson(dbContext);
+  }
+
+  public static AwsCloudContextFields deserialize(String contextJson) {
+    try {
+      AwsCloudContextV1 dbContext = DbSerDes.fromJson(contextJson, AwsCloudContextV1.class);
+      dbContext.validateVersion();
+
+      return new AwsCloudContextFields(
+          dbContext.majorVersion,
+          dbContext.organizationId,
+          dbContext.accountId,
+          dbContext.tenantAlias,
+          dbContext.environmentAlias);
+
+    } catch (SerializationException e) {
+      throw new InvalidSerializedVersionException("Invalid serialized version", e);
+    }
+  }
+
+  /** Mask AWS version numbers so as not to collide with Azure and GCP version numbers */
```

Comment:
i don't understand the purpose of this mask..but i see you are just moving files around.

Reply:
This is equivalent to saying the version is 257. I don't know why it was done this way.